deps/thpool/thpool.c

```diff
@@ -293,89 +296,61 @@ static void redisearch_thpool_verify_init(struct redisearch_thpool_t *thpool_p)
   }
 
   /* Add new threads if needed */
-  for (size_t n = 0; n < n_new_threads; n++) {
-    thread_init(thpool_p);
-  }
+  if (n_new_threads > 0) {
+    bool started[n_new_threads];
```

Comment:
Shouldn't this variable also be set as volatile? So the waiting thread actually checks the value, and it is not optimized away?

Reply:
Also, is there a reason to use a flag array instead of an atomic counter?